### ISSUE TITLE
glibc: support option arguments in getopt_long

### ISSUE THREAD

**Issue by Dennisbonke** [managarm/mlibc#170]
On managarm, `xbps` triggers the following assertion

```
In function getopt_long, file ../../../src/mlibc/options/glibc/generic/getopt-stubs.cpp:86
__ensure(
(opt[1] != ':') && "We do not support option arguments.
```

This should be implemented for `xbps` support. Furthermore, it is required by several programs from `shadow`

**Comment by Dennisbonke** [comment]
also, running `su - managarm` triggers the following on managarm
```
In function getopt_long, file ../../../src/mlibc/options/glibc/generic/getopt-stubs.cpp:92
__ensure(
(strlen(argv[optind]) == 2)
```

**Comment by avdgrinten** [comment]
This issue should probably be prioritized. It would also be nice to somehow unify the implementations of getopt and getopt_long.

**Comment by Dennisbonke** [comment]
Several programs from `util-linux`, most notably `mount`, will not take short options without this issue resolved.

**Comment by Geertiebear** [comment]
Closed by #438 .
